romancal/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,185 @@
+
+"""
+
+
+def create_astrometric_catalog(input_models, catalog="GAIADR2", output="ref_cat.ecsv",
+                               gaia_only=False, table_format="ascii.ecsv",
+                               existing_wcs=None, num_sources=None):
+    """Create an astrometric catalog that covers the inputs' field-of-view.
+
+    Parameters
+    ----------
+    input : str, list
+        Filenames of images to be aligned to astrometric catalog
+
+    catalog : str, optional
+        Name of catalog to extract astrometric positions for sources in the
+        input images' field-of-view. Default: GAIADR2. Options available are
+        documented on the catalog web page.
+
+    output : str, optional
+        Filename to give to the astrometric catalog read in from the master
+        catalog web service.  If None, no file will be written out.
+
+    gaia_only : bool, optional
+        Specify whether or not to only use sources from GAIA in output catalog
+
+    existing_wcs : model
+        existing WCS object specified by the user as generated by
+        `resample.resample_utils.make_output_wcs`
+
+    num_sources : int
+        Maximum number of brightest/faintest sources to return in catalog.
+        If `num_sources` is negative, return that number of the faintest
+        sources.  By default, all sources are returned.
+
+    Notes
+    -----
+    This function will point to astrometric catalog web service defined
+    through the use of the ASTROMETRIC_CATALOG_URL environment variable.
+
+    Returns
+    -------
+    ref_table : `~astropy.table.Table`
+        Astropy Table object of the catalog
+
+    """
+
+    # start by creating a composite field-of-view for all inputs
+    # This default output WCS will have the same plate-scale and orientation
+    # as the first member in the list.
+    # Fortunately, for alignment, this doesn't matter since no resampling of
+    # data will be performed.
+    if existing_wcs is not None:
+        outwcs = existing_wcs
+    else:
+        outwcs = resample_utils.make_output_wcs(input_models)
+    radius, fiducial = compute_radius(outwcs)
+
+    # perform query for this field-of-view
+    ref_dict = get_catalog(fiducial[0], fiducial[1], sr=radius, catalog=catalog)
+    colnames = ('ra', 'dec', 'mag', 'objID')
+
+    ref_table = ref_dict[colnames]
+
+    # Add catalog name as meta data
+    ref_table.meta['catalog'] = catalog
+    ref_table.meta['gaia_only'] = gaia_only
+
+    # rename coordinate columns to be consistent with tweakwcs
+    ref_table.rename_column('ra', 'RA')
+    ref_table.rename_column('dec', 'DEC')
+
+    # Append GAIA ID as a new column to the table...
+    gaia_sources = []
+    for source in ref_dict:
+        if 'GAIAsourceID' in source:
+            g = source['GAIAsourceID']
+            if gaia_only and g.strip() == '':
+                continue
+        else:
+            g = "-1"  # indicator for no source ID extracted
```

Comment:
If no GAIA sources are found should the code issue a warning? an info message?

Reply:
The check for the number of sources found within the search area occurs after returning to the caller (on line 391 of `tweakreg_step.py`):
```# Check that there are enough GAIA sources for a reliable/valid fit
 num_ref = len(ref_cat)
 if num_ref < self.abs_minobj:
    # Raise Exception here to avoid rest of code in this try block
    self.log.warning(
        f"Not enough sources ({num_ref}) in the reference catalog "
        "for the single-group alignment step to perform a fit. "
        f"Skipping alignment to the {self.abs_refcat} reference "
        "catalog!"
    )```